vivisect/impemu/emulator.py

```diff
@@ -50,14 +49,20 @@ def __init__(self, vw, logwrite=False, logread=False):
 
         self.stack_map_mask = e_bits.sign_extend(0xfff00000, 4, vw.psize)
         self.stack_map_base = e_bits.sign_extend(0xbfb00000, 4, vw.psize)
-        self.stack_pointer = self.stack_map_base + 4096
+        self.stack_pointer = self.stack_map_base + stacksize
 
         # Possibly need an "options" API?
         self._safe_mem = True   # Should we be forgiving about memory accesses?
         self._func_only = True  # is this emulator meant to stay in one function scope?
 
         # Map in a memory map for the stack
-        self.addMemoryMap(self.stack_map_base, 6, "[stack]", init_stack_map)
+        stack_map = init_stack_map
+        if stacksize != 4096:
+            stack_map = ''
+            for i in xrange(stacksize):
```

Comment:
Trying to recall for sure, but i thought that the emulator ended up setting sp to the somwhere within the stack map rather than all the way at the bottom.  ( maybe there was a hard-coded 8192 somewhere that I'm missing... memory is fuzzy this AM ;) )